packages/middleware-retry/src/retryMiddleware.ts

```diff
@@ -55,7 +56,7 @@ export const retryMiddleware = (options: RetryResolvedConfig) => <Output extends
         output.$metadata.attempts = attempts + 1;
         output.$metadata.totalRetryDelay = totalRetryDelay;
         return { response, output };
-      } catch (e) {
+      } catch (e: any) {
```

Comment:
nit: do we need to cast this as `any`?

Reply:
without the type annotation this was `any` already, I made it explicit.

Alternative is to set `unknown` and cast below to SdkError when referencing `e`.